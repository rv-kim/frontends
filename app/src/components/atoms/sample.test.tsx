import * as React from 'react'
import { shallow } from 'enzyme'
import * as Renderer from 'react-test-renderer'
import Sample from './sample'

describe('Sample Component', () => {

  test('snapshot', () => {
    const ss = Renderer.create(<Sample text="test" color="black" />).toJSON()
    expect(ss).toMatchSnapshot()
  })

  test('スナップショット０２', () => {
    const ss = Renderer.create(<Sample text="examples" color="green" />).toJSON()
    expect(ss).toMatchSnapshot()
  })

  test('H1 should be "Hello, World."', () => {
    const sample = shallow(<Sample />)
    expect(sample.find('h1').text()).toBe('Hello World.')
  })

  test('H2 should be "test..."', () => {
    const sample = shallow(<Sample />)
    expect(sample.find('h2').text()).toBe('test...')
  })

  test('H3 should be text property', () => {
    const txt = 'halo, halo'
    const sample = shallow(<Sample text={txt} />)
    expect(sample.find('h3').text()).toBe(txt)
  })

})
