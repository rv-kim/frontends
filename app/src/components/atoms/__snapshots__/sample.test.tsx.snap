// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Component snapshot 1`] = `
Array [
  <h1>
    Hello World.
  </h1>,
  <h2>
    test...
  </h2>,
  <h3>
    test
  </h3>,
  <style
    jsx={true}
  >
    
          h1 {color: black}
          h2 {color: blue}
          h3 {color: black}
        
  </style>,
]
`;

exports[`Sample Component スナップショット０２ 1`] = `
Array [
  <h1>
    Hello World.
  </h1>,
  <h2>
    test...
  </h2>,
  <h3>
    examples
  </h3>,
  <style
    jsx={true}
  >
    
          h1 {color: black}
          h2 {color: blue}
          h3 {color: green}
        
  </style>,
]
`;
